Improve slurm job scheduler packaging

adevress opened the pull request:
###### Motivation for this change
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
- add support for numa and hwloc
- add support for lua scripting module
- make slurm compatible with system configuration by default, in case of  Nix used
  as a side package manager (tested on redhat)
- strip the libtool generated files
- add possibility to add system specific modules

jagajaga commented on pkgs/servers/computing/slurm/default.nix:
```diff
@@ -10,18 +10,26 @@ stdenv.mkDerivation rec {
   configureFlags =
     [ "--with-munge=${munge}"
       "--with-ssl=${openssl}"
+      "--sysconfdir=/etc/slurm"
```
I think it should be  just `/etc`

adevress replied:
It should but it is not :)
In fact, all the main distribution including the official [slurm spec file ](https://github.com/SchedMD/slurm/blob/master/slurm.spec), RedHat, Fedora and Debian override the %sysconfdir to /etc/slurm due to the quite large numbers of conf files of slurm.

jagajaga commented on pkgs/servers/computing/slurm/default.nix:
```diff
@@ -1,5 +1,5 @@
 { stdenv, fetchurl, pkgconfig, curl, python, munge, perl, pam, openssl,
-  ncurses, mysql, gtk }:
+  ncurses, mysql, gtk, lua, hwloc, numactl, extraDeps ? [] }:
```
Why `extraDeps`?

lancelotsix replied:
What is the expected use of the `extraDeps` parameter ?

I think it would make sense to have explicit (and null-able) parameters for any additional build input

adevress replied:
Mainly to support arbitrary plugins like auks or super-computer ones with a simple override.

extraDeps is passed as a buildInput

adevress replied:
But tell me if it is a blocking issue

joachifm replied:
@adevress is it infeasible to encode optional dependencies explicitly as optional parameters? I think at least `extraDeps` needs a comment explaining what the expected value is, it's not a good interface currently, IMO.

adevress commented:
@joachifm Fine for me. I edited it and removed the extraDeps. 
I will propose a separate pull request for auks / lua support if I have time.

Adev

adevress commented:
ping @joachifm @jagajaga  :)

joachifm commented:
Needs rebasing.

Looks to me like `.la` files are typically not removed, are there any negative consequences to doing so here?

lancelotsix commented:
I do not think any-one will need them. Those are libtool files that can be used to use a library (if nothing such as pkg-config files is provided).

I think `slurm` is used for its binaries, not libraries. There is no big point in keeping those files (and they are not necessary if someone needs to link against `libslurm.so`). On the other hand, they do no harm, except for the few bytes the use on disks.

By the way, many distros tend to remove them, even for libraries (see https://wiki.debian.org/ReleaseGoals/LAFileRemoval for debian, https://fedoraproject.org/wiki/Packaging:Guidelines for fedora). I do not know any guideline in Nixpkgs about those, packages usually do not take care about then, so they are kept.

joachifm commented:
@lancelotsix makes sense. I think it'd be good to approach this more systematically rather than on a per-package basis, though ... then again, if they're effectively left-over cruft in this case then I guess it's okay to delete them.

adevress committed:
Improve slurm job scheduler packaging

    - add support for numa and hwloc
    - add support for lua scripting module
    - make slurm compatible with system configuration by default, in case of  Nix used
      as a side package manager (tested on redhat)
    - strip the libtool generated files

adevress commented:
@joachifm I did the rebase

@joachifm  @lancelotsix Yes they are libtool files, they are only used by libtool for "embedded builds" and useless in this configuration. Most of the Linux distributions normally strip them off. In this precise case, I remove them explicitly because they can generate some conflicts with buggy slurm lua plugins and do not bring any benefits.
